confluent_kafka/avro/__init__.py

```diff
@@ -102,23 +102,23 @@ class AvroConsumer(Consumer):
     """
     def __init__(self, config, schema_registry=None):
 
-        schema_registry_url = config.pop("schema.registry.url", None)
-        schema_registry_ca_location = config.pop("schema.registry.ssl.ca.location", None)
-        schema_registry_certificate_location = config.pop("schema.registry.ssl.certificate.location", None)
-        schema_registry_key_location = config.pop("schema.registry.ssl.key.location", None)
+        sr_conf = {key.replace("schema.registry.", ""): value
+                   for key, value in config.items() if key.startswith("schema.registry.")}
+
+        if config.get("schema.registry.basic.auth.credentials.source") == 'SASL_INHERIT':
+            sr_conf['sasl.mechanisms'] = config.get('sasl.mechanisms', '')
```

Comment:
There's a mix of `'` and `"` here. I do the same, I just wanted to point it out that we're the same, and it looks fantastic!

Reply:
Yeah I'm trying to come up with some sort of convention since pep8 fails to make one. As you can tell I'm not sure which way to go